### PR TITLE
Make huggingface[cli]  and omlmd optional

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/ramalama/issues/211

### CODE REVIEW

**Comment by lsm5** [comment]
as long as installation from the packit job rpm works, lgtm from me. I'll try this out in a bit.

**Review by lsm5 — APPROVED** [review]
rpm update with packit rpm works fine

```
sudo dnf install -y dnf-plugins-core
sudo dnf copr enable packit/containers-ramalama-212
sudo dnf install -y python3-ramalama-0-1.20241001151639838794.pr212.fc42.noarch
```

LGTM

**Comment by rhatdan** [comment]
These requires are coming from the pyproject.toml, @lsm5 do you know if there is a way we could add them there but remove them from the spec requires.

It would be nice if we could get them automatically when people install via PyPi

File: ramalama/huggingface.py [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import os
-from ramalama.common import run_cmd, exec_cmd
+from ramalama.common import run_cmd, exec_cmd, perror
 from ramalama.model import Model
 
+missing_hugginface="""
+Huggingface models requires the huggingface-cli and tldm modules.
+These modules can be installed via:
+pip install huggingface_hub[cli] tldm
```
Comment (ericcurtin) [inline ramalama/huggingface.py:8]: I don't think we need tldm, huggingface_hub[cli] should be fine

Comment (ericcurtin) [reply ramalama/huggingface.py:8]:
```
huggingface-cli: command not found
```

would have been fine too. A user may want to install via pip, pip3, pipx, dnf, apt. Maybe we should let them decide how to install. Some people don't recommend pip.

Comment (rhatdan) [reply ramalama/huggingface.py:8]: The issue is we don't know if those packages exists via other means.  I can add, 
`or modules might be available via pip3, pipx, dnf, apt or other install tools.`

**Review by ericcurtin — APPROVED** [review]

File: ramalama/huggingface.py (hunk 1 above)
Comment (ericcurtin) [reply ramalama/huggingface.py:9]: approved nonetheless :)

**Pull request merged**